neutron/CMakeLists.txt

```diff
@@ -1,19 +1,36 @@
 # SPDX-License-Identifier: GPL-3.0-or-later
 # Copyright (c) 2019 Scipp contributors (https://github.com/scipp)
+set(TARGET_NAME "scipp-neutron")
 set(INC_FILES include/scipp/neutron/convert.h)
 
 set(SRC_FILES convert.cpp
   )
 
-add_library(scipp-neutron STATIC ${INC_FILES} ${SRC_FILES})
-target_link_libraries(scipp-neutron PUBLIC scipp-core)
+set(LINK_TYPE "STATIC")
+if(DYNAMIC_LIB)
+  set(LINK_TYPE "SHARED")
+endif(DYNAMIC_LIB)
+
+add_library(${TARGET_NAME} ${LINK_TYPE} ${INC_FILES} ${SRC_FILES})
+generate_export_header(${TARGET_NAME})
+target_link_libraries(${TARGET_NAME} PUBLIC scipp-core)
 target_include_directories(
-  scipp-neutron
+  ${TARGET_NAME}
   PUBLIC $<INSTALL_INTERFACE:include>
-         $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>)
-target_include_directories(scipp-neutron SYSTEM
+         $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+         ${CMAKE_CURRENT_BINARY_DIR})
+target_include_directories(${TARGET_NAME} SYSTEM
                            PUBLIC ${CMAKE_BINARY_DIR}/Eigen-src
                            PRIVATE "../range-v3/include")
-set_target_properties(scipp-neutron PROPERTIES POSITION_INDEPENDENT_CODE TRUE)
+
+if(NOT ENABLE_LTO)
+    set_property(TARGET ${TARGET_NAME} PROPERTY INTERPROCEDURAL_OPTIMIZATION FALSE)
+endif(NOT ENABLE_LTO)
```

Comment:
Instead of doing this in every module, change it globally here: https://github.com/scipp/scipp/blob/master/CMakeLists.txt#L57

Reply:
I want the LTO disabled for tests forever and a flag for libraries. So either you would do this manually for tests or for libs, right?

Reply:
Yes, keep it manually disabled for tests, everything else using the global cmake flag.

Reply:
I can remove this flag and we can use CMAKE_INTERPROCEDURAL_OPTIMIZATION wich also affects producing the fat lto objects, but all the optimization would be ignored with -flto switched off.

Reply:
Do we care about fat lto objects?